web/docs/auth/email.md

```diff
@@ -1,3 +1,1011 @@
+    </Layout>
+  );
+}
+
+// A layout component to center the content
+export function Layout({ children }: { children: React.ReactNode }) {
+  return (
+    <div className="w-full h-full bg-white">
+      <div className="min-w-full min-h-[75vh] flex items-center justify-center">
+        <div className="w-full h-full max-w-sm p-5 bg-white">
+          <div>{children}</div>
+        </div>
+      </div>
+    </div>
+  );
+}
+```
+</TabItem>
+</Tabs>
+
+We imported the generated Auth UI components and used them in our pages. Read more about the Auth UI components [here](/docs/auth/ui).
+
+### 5. Set up an email sender
+
+To support e-mail verification and password reset flows, we need an e-mail sender. Luckily, Wasp supports several email providers out of the box.
+
+We'll use SendGrid in this guide to send our e-mails. You can use any of the supported email providers.
+
+To set up SendGrid to send emails, we will add the following to our `main.wasp` file:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+app myApp {
+  // ...
+  // 7. Set up the email sender
+  emailSender: {
+    provider: SendGrid,
+  }
+}
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp"
+app myApp {
+  // ...
+  // 7. Set up the email sender
+  emailSender: {
+    provider: SendGrid,
+  }
+}
+```
+</TabItem>
+</Tabs>
+
+... and add the following to our `.env.server` file:
+
+```c title=".env.server"
+SENDGRID_API_KEY=<your key>
+```
+
+If you are not sure how to get a SendGrid API key, read more [here](/docs/advanced/email#getting-the-api-key).
+
+Read more about setting up email senders in the [sending emails docs](/docs/advanced/email).
+
+### Conclusion
+
+That's it! We have set up email authentication in our app. 🎉
+
+Running `wasp db migrate-dev` and then `wasp start` should give you a working app with email authentication. If you want to put some of the pages behind authentication, read the [using auth docs](/docs/auth/overview).
+
+## Login and signup flows
+
+### Login
+
+![Auth UI](/img/authui/login.png)
+
+If logging in with an unverified email is allowed, the user will be able to login with an unverified email address. If logging in with an unverified email is not allowed, the user will be shown an error message.
+
+Read more about the `allowUnverifiedLogin` option [here](#allowunverifiedlogin-bool-specifies-whether-the-user-can-login-without-verifying-their-e-mail-address).
+
+### Signup
+
+![Auth UI](/img/authui/signup.png)
+
+Some of the behavior you get out of the box:
+1. Rate limiting
+  
+  We are limiting the rate of sign-up requests to **1 request per minute** per email address. This is done to prevent spamming.
+2. Preventing user email leaks
+  
+  If somebody tries to signup with an email that already exists and it's verified, we'll give back the same response as if the user registered successfully. This is done to prevent leaking the user's email address.
```

Comment:
If somebody tries to create an account with an `email` that exists and that the email is verified:
- instead of saying "hey this is an existing email"
- we say "hey the account is created, please verify your email"

This is the same behaviour as if the account is created, but no new account is created.